src/lib/rank_utils.ts

```diff
@@ -22,9 +22,74 @@ interface IRankInfo {
     label: string;
 }
 
-export const MaxRank: number = 36;
+class Rating {
+    unset:boolean;
+    rating:number;
+    deviation:number;
+    volatility:number;
+    rank:number;
+    rank_label:string;
+    partial_rank:number;
+    partial_rank_label:string;
+    rank_deviation_labels:Array<string>;
+    rank_deviation:number;
+}
+
+export const MaxRank: number = 39;
+
+
+const MIN_RATING = 100;
+const MAX_RATING = 6000;
+
+export function rank_to_rating(rank:number) {
+    return 900 * Math.exp(0.032 * rank);
+}
+
+export function rating_to_rank(rating:number) {
+    return Math.log(Math.min(MAX_RATING, Math.max(MIN_RATING, rating)) / 900.0) / 0.032;
+}
+
+export function get_handicap_adjustment(rating:number, handicap:number):number {
+    return rank_to_rating(rating_to_rank(rating) + handicap) - rating;
```

Comment:
Take the 1d no komi case. Black wins either 39.4% or 39.2% of the time depending on your variables. If you use "H", you are telling Glicko2 that Black should win 50% of the time. If you use "H-0.5" you are telling Glicko2 that Black should win something like 44.5% of the time. So H-0.5 seems to model reality much better.

Suppose you use H, and a 1d and 1k play many H1 games. Glicko2 will see the 1k only wins 39% of the time, and reduce his rating quite a lot. 

If you do the same thing using H-0.5, again Glicko2 sees the 1k winning 39% of the time, but now since the expectation is only 44.5%, Glicko2 will not penalize the 1k as much.